Architecture.md

```diff
@@ -71,6 +71,116 @@
+
+Here's an overview of the interactions of the git controller:
+
+```mermaid
+sequenceDiagram
+    participant API
+    participant GitRepoManager
+    participant HostMachine
+
+    API ->> GitRepoManager: CreateProject / DeleteProject
+    GitRepoManager ->> HostMachine: git init / rm -rf project_id
+    HostMachine -->> GitRepoManager: Ok
+    GitRepoManager -->> API: Ok
+```
+
+##### Authentication
+
+Once the repositories are stored, the clients need to be able to push to them.
+
+In order to let the users push to thoses repositories, we need to give them access to the host machine. An easy solution would be to give them a ssh access to the host machine with servers like [sshd](https://www.openssh.com/). However, this would give them access to the whole host machine, which is not what we want. Indeed, we want to limit the access of the clients to their own repositories, and not to the whole host machine.
+
+One solution would be to use a git server like [gitea](https://gitea.io/en-us/) or [gitlab](https://about.gitlab.com/), which provide a full-fledged git service, with authentication and authorization. However, this would require to maintain a full git server, with possibly duplicated data, which is not the goal of PaaSTech.
+
+The option we chose is to create a custom git server, which handles the authentication and authorization of the clients, and then execute only the command `git-receive-pack` which is used to push to a repository. This way, the clients can only push to their own repositories, and not to the whole host machine.
+
+The advantage of this solution is that it is lightweight and give us full control over the authentication and authorization of the clients. It also allows us to use the same authentication and authorization system as the rest of the application, which is easier to maintain.
+
+The git server is written in golang, for its simplicity and its performance. It is also well documented, and has a lot of libraries to help us build the server, unlike Rust which was the original choice. The libraries used are [ssh made by gliderlabs](https://github.com/gliderlabs/ssh) which is used to create git clients and servers. It is also used by [gitea](https://about.gitea.com) to create their git server.
+
+The authentication and authorization process is the following:
+
+- The client connects to the git server using ssh via `git push`
+- The git server checks if the client is authenticated using the ssh key provided by the client by matching it against the ssh keys stored in the database
+    - If the client is not authenticated, the connection is closed
+- The git server checks if the client is authorized to push to the repository by checking if the client is the owner of the repository
+    - If the client is not authorized, the connection is closed
+- The git server executes the command `git-receive-pack` which is used to push to a repository
```

Comment:
Please use a numbered list instead of a bulleted list when describing steps to follow in order